### PR TITLE
[DSLX:LS] Get go-to-definition working across files and add a test.

### PR DESCRIPTION
**Pull request by cdleary**
NOTE: do not land until google/xls#1563 lands, this is a dependent PR.

### CODE REVIEW

File: xls/dslx/lsp/language_server_adapter.cc [hunk 1]
```diff
@@ -82,14 +83,54 @@ void AppendDiagnosticFromTypecheck(
...
 
+// If the span's filename is a relative path, we map it into a URI by finding
+// its path on disk.
+//
+// Note: there may be a more holistic way to do this long term -- the LSP
+// effectively requires us to create a filesystem overlay that mixes real files
+// and in-memory files with modifications. For now, this at least enables
+// things like go-to-definition in files that have not yet been loaded into the
+// language server.
+absl::StatusOr<std::string> MaybeRelpathToUri(
```
Comment (hzeller) [inline xls/dslx/lsp/language_server_adapter.cc:94]: There are also some path/uri functions in [`verible/common/lsp/lsp-file-utils.h`](https://github.com/chipsalliance/verible/blob/master/common/lsp/lsp-file-utils.h), though probably not exactly what you need here. Mostly an FYI.

**Review by hzeller — APPROVED** [review]
LGTM

File: xls/dslx/lsp/language_server_adapter.cc (hunk 1 above)
Comment (cdleary) [reply xls/dslx/lsp/language_server_adapter.cc:94]: Thanks, good to know -- if I see more actually-URI-looking-encoded stuff those look helpful.

**Comment by hzeller** [comment]
This might need a rebase.

**Comment by hongted** [comment]
@cdleary -- Has @hzeller it looks like there's a branch conflict that's preventing this from being merged.  Can you rebase/update the PR?

**Comment by hongted** [comment]
Have created a rebased PR on https://github.com/google/xls/pull/1577

**Comment by cdleary** [comment]
Thanks @hongted -- sorry I fell behind on checking GH inbox last week.

**Pull request closed without merging**